backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/service/impl/InterimInvoiceFlatrateTermBL.java

```diff
@@ -0,0 +1,192 @@
+				.map(inOutLine -> Quantitys.create(inOutLine.getMovementQty(), UomId.ofRepoId(inOutLine.getC_UOM_ID())))
+				.reduce(Quantitys.createZero(interimInvoiceFlatrateTerm.getUomId()), Quantity::add);
+	}
+
+	@Override
+	public void updateInvoicedQtyForPartialPayment(@NonNull final I_C_Invoice_Candidate invoiceCand)
+	{
+		if (!invoiceCand.isInterimInvoice())
+		{
+			return;
+		}
+		final InvoiceCandidateId icId = InvoiceCandidateId.ofRepoId(invoiceCand.getC_Invoice_Candidate_ID());
+		final InterimInvoiceFlatrateTerm interimInvoiceFlatrateTerm = interimInvoiceFlatrateTermDAO.getInterimInvoiceFlatrateTermForWithwoldingOrInterimICId(icId);
+		if (interimInvoiceFlatrateTerm == null)
+		{
+			Loggables.withLogger(logger, Level.WARN).addLog("Invoice candidate with ID={} marked as partial payment, but no InterimInvoiceFlatrateTerm record found for it.", icId);
+			return;
+		}
+		interimInvoiceFlatrateTermDAO.save(interimInvoiceFlatrateTerm.toBuilder()
+				.qtyInvoiced(invoiceCand.getQtyInvoiced())
+				.build());
+		interimInvoiceFlatrateTermLineDAO.setInvoiceLineToLines(InvoiceCandidateIds.ofRecord(invoiceCand), interimInvoiceFlatrateTerm.getId());
+	}
+
+	/**
+	 * Update the interim and withholding ICs of an {@link InterimInvoiceFlatrateTerm} for the given {@link I_M_InOutLine}.
+	 */
+	private void updateInterimInvoiceOverview(@NonNull final InterimInvoiceFlatrateTerm interimInvoiceFlatrateTerm, @NonNull final I_M_InOutLine inOutLine)
+	{
+		final InOutAndLineId inOutAndLineId = InOutAndLineId.ofRepoId(inOutLine.getM_InOut_ID(), inOutLine.getM_InOutLine_ID());
+		if (interimInvoiceFlatrateTermLineDAO.getByInOutLineId(inOutAndLineId.getInOutLineId()) != null)
+		{ //maybe it's already associated
+			return;
+		}
+
+		interimInvoiceFlatrateTermLineDAO.createInterimInvoiceLine(interimInvoiceFlatrateTerm, inOutAndLineId);
+
+		updateQuantities(interimInvoiceFlatrateTerm);
+	}
+
+	private void setICQtyies(final I_C_Invoice_Candidate invoiceCandidate, final Quantity qtyToInvoice)
+	{
+		final BigDecimal qtyDeliveredInICUOM = qtyToInvoice.toBigDecimal();
+		invoiceCandidate.setQtyOrdered(qtyDeliveredInICUOM);
+		invoiceCandidate.setQtyEntered(qtyDeliveredInICUOM);
+		invoiceCandidate.setQtyDelivered(qtyDeliveredInICUOM);
+		invoiceCandidate.setQtyDeliveredInUOM(qtyDeliveredInICUOM);
+		invoiceCandidate.setQtyToInvoice(qtyDeliveredInICUOM);
+		invoiceCandidate.setQtyToInvoiceInUOM(qtyDeliveredInICUOM);
+		invoiceCandidate.setQtyToInvoiceBeforeDiscount(qtyDeliveredInICUOM);
+		invoiceCandDAO.save(invoiceCandidate);
+	}
+
+	private InterimInvoiceFlatrateTerm createInterimInvoiceFlatrateTerm(@NonNull final InterimInvoiceFlatrateTerm interimInvoiceFlatrateTerm, final @NonNull I_M_InOutLine inOutLine)
+	{
+		final I_C_Flatrate_Term flatrateTerm = flatrateDAO.retrieveTerm(interimInvoiceFlatrateTerm.getFlatrateTermId());
+
+		return InterimInvoiceFlatrateTermCreateCommand.builder()
+				.ctx(InterfaceWrapperHelper.getContextAware(inOutLine).getCtx())
```

Comment:
why ctx is needed?
pls try to avoid using ctx

Reply:
removed